include/engine/guidance/post_processing.hpp

```diff
@@ -28,6 +28,7 @@ std::vector<RouteStep> collapseTurns(std::vector<RouteStep> steps);
 
 // A check whether two instructions can be treated as one. This is only the case for very short
 // maneuvers that can, in some form, be seen as one. Lookahead of one step.
+// This is only a pre-check and does not necessarily allow collapsing turns!!!
```

Comment:
well. its a precondition, I can see what we can do here. In my post-processing refactor, this should be already better, though.
One function over there might suffer from the same miss-readability. Will change https://github.com/Project-OSRM/osrm-backend/pull/3495/files#diff-2a61b83618a06f087b5d5f03384829ceR58, since this function here is not going to survive for long.